### DISCUSSION TITLE
Passing additional data to event handlers - rio.Switch

### DISCUSSION THREAD

**Discussion by Nachsh** [rio-labs/rio#194] [Q&A] [answered]
I managed to use the "partial" method with rio.Button, as given in the documentation.

`    def build(self) -> rio.Component:
        return rio.Column(
            rio.Button(
                "First Button",
                on_press=functools.partial(self.on_button_press, "Button 1"),
            )`

but what I really need is to use it with rio.Switch. I tried to use this syntax on Switch and failed. what am I missing?

**Comment by Aran-Fey** [comment]
It should work the same with a `Switch`. Please show us the code that's not working.

**Reply by Nachsh** [reply, marked as answer]
@Aran-Fey I got some confidence from your answer and I tried again. It works now!
my intake is that I need to place the 'event' last in the call for the 'on_change' function:

call:
on_change=functools.partial(self.on_led_change, curr_led)

func def:
def on_led_change(self, led: Led, event: rio.SwitchChangeEvent ):

Thanks!
